src/components/playback/playbackmanager.js

```diff
@@ -3296,18 +3296,21 @@ class PlaybackManager {
             const streamInfo = error.streamInfo || getPlayerData(player).streamInfo;
 
             if (streamInfo?.url) {
+                const isAlreadyFallbacking = streamInfo.url.toLowerCase().includes('transcodereasons=directplayerror');
```

Comment:
`transcodereasons=directplayerror` doesn't work because it happens when StreamBuilder didn't find the DirectProfile.

Tested by hacking `htmlVideoPlayer` and it is looping (4 tries are expected):
```diff
diff --git a/src/plugins/htmlVideoPlayer/plugin.js b/src/plugins/htmlVideoPlayer/plugin.js
index f26e89cc9..adbd7675c 100644
--- a/src/plugins/htmlVideoPlayer/plugin.js
+++ b/src/plugins/htmlVideoPlayer/plugin.js
@@ -391,7 +391,12 @@ export class HtmlVideoPlayer {
 
         return this.createMediaElement(options).then(elem => {
             return this.updateVideoUrl(options).then(() => {
-                return this.setCurrentSrc(elem, options);
+                return this.setCurrentSrc(elem, options).then((r) => {
+                    setTimeout(() => {
+                        if (this.#mediaElement) onErrorInternal(this, MediaError.MEDIA_DECODE_ERROR);
+                    }, 3000);
+                    return r;
+                });
             });
         });
     }
```

I think we should test for `transcodereasons` because `transcodereasons` is an indication of performing "transcoding" (remux, direct stream).
```suggestion
                const isAlreadyFallbacking = streamInfo.url.toLowerCase().includes('transcodereasons');
```